src/formatDate.js

```diff
@@ -50,7 +50,41 @@
  */
 
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const oldFormat = {
+    getYear(YY) {
+      if (this.YY < 30) {
+        this.YYYY = '20' + this.YY;
+      } else {
+        this.YYYY = '19' + this.YY;
+      }
+    },
```

Comment:
You can store old format like `Y: 'YY'`cand move the logic of getting year from loop in newFormat to this method, with all the conditions